### DISCUSSION TITLE
Testing the application of slice-timing correction over the right axis and direction

### DISCUSSION THREAD

**Discussion by Gab-D-G** [CoBrALab/RABIES#217] [Show and tell]
After conversion to RAS, the Y dimension should correspond to anteroposterior slices, where the slices are usually acquired from front to back. AFNI's 3dTshift applies correction over the Z axis, since in humans slices are acquired over the superior-inferior axis. Thus, axes are temporally swapped within RABIES to match the desired axis for correction (axis Y by default) with the Z axis.

However, in the anteroposterior axis, the index order is inverted where the first index is at the back of the brain and the last index is the front of the brain. 
The first slice acquired in the olfactory bulb -> corresponds to the last index in the data array
![front_slice](https://user-images.githubusercontent.com/34081874/161813703-9559719e-9b1e-423a-aa6d-fb7b8ed2f478.png)
The last slice acquired in the cerebellum -> corresponds to the first index in the data array
![back_slice](https://user-images.githubusercontent.com/34081874/161813638-399571a2-394e-4487-82ca-b1853176e86c.png)

It is unclear in which direction 3dTshift will be applying the correction, so here I am testing its application.

First, using a negative direction with `-tpattern seq-z`, we can observe in a voxel from the front slice that with the correction (shown in orange) the data is shifted relative to the original timecourse (in blue) to be more similar to the next timepoint. This means that the interpolation was applied to shift the slice timing to be **later in time**:
![front_shift](https://user-images.githubusercontent.com/34081874/161814568-725cb16b-962b-4104-b959-2766ac5c175d.png)

Conversely, in a voxel from the last slice,  the data is shifted to be more similar to the last timepoint, which means that the timing was shifted to be **earlier in time**:
![back_shift](https://user-images.githubusercontent.com/34081874/161817102-9a64b1f9-d9be-4468-bdb0-af0cb69e8890.png)

This is the outcome desired if slices were acquired from front to back, thus the default in RABIES is set to be with a negative direction. However, the correction can be applied in the positive direction (from back to front) by specifying the `--tpattern seq+z` instead when running RABIES.

If we look at a voxel in the middle slice, we can see that no correction was applied (the corrected orange trace is overlapping with the blue trace):
![middle_trace](https://user-images.githubusercontent.com/34081874/161821944-b8018ef2-0994-482a-b51e-72e4340447e4.png)
From the slice:
![middle_slice](https://user-images.githubusercontent.com/34081874/161822085-3454af75-804b-4c28-b696-c9a527c5bd78.png)

This confirms that the correction was applied over the anteroposterior axis.

**Comment by gdevenyi** [comment]
Ideally the acquisition direction should be parsed from the BIDS input data, I believe its required in the metadata?

**Reply by Gab-D-G** [reply]
I am not sure how common this information is, there are definitely cases where it is lacking. Sometimes the TR isn't even part of the metadata, or the data itself can be artificially modified (e.g. axes swapped), which requires flexibility with the parameter settings.

**Reply by gdevenyi** [reply]
You can start by using pybids to query `SliceEncodingDirection`[1] and allow overrides.

[1] https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html

**Reply by Gab-D-G** [reply]
that can be raised as an issue.
